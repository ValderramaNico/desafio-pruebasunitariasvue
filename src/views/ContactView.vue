<template>
  <div class="contact">
    <h1>PÃ¡gina "contacto"</h1>
  </div>
</template>
