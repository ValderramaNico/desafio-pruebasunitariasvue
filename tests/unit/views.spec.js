import { mount } from "@vue/test-utils";
import ContactView from "@/views/ContactView.vue";
import AboutView from "@/views/AboutView.vue";
import { createRouter, createWebHistory } from "vue-router";

describe("Test vista 'About'", () => {
  it("Test1 - Snapshot vista 'About' succesful", () => {
    const wrapper = mount(AboutView);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Test2 - Prueba existencia del componente en la ruta", async () => {
    const routes = [
      {
        path: "/about",
        name: "about",
        component: () => AboutView,
      },
    ];
    const router = createRouter({
      history: createWebHistory(process.env.BASE_URL),
      routes,
    });

    router.push("/about");
    await router.isReady();

    const wrapper = mount(AboutView, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.findComponent(AboutView).exists()).toBe(true);
  });
});

describe("Tests de vista 'Contact'", () => {
  it("Test1 - Snapshot vista 'Contact'", () => {
    const wrapper = mount(ContactView);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Test2 - Prueba existencia del componente en la ruta", async () => {
    const routes = [
      {
        path: "/contact",
        name: "contact",
        component: () => ContactView,
      },
    ];
    const router = createRouter({
      history: createWebHistory(process.env.BASE_URL),
      routes,
    });

    router.push("/contact");
    await router.isReady();

    const wrapper = mount(ContactView, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.findComponent(ContactView).exists()).toBe(true);
  });
});
