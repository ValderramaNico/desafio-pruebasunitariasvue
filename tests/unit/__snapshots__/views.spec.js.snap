// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test vista 'About' Test1 - Snapshot vista 'About' succesful 1`] = `
<div class="about">
  <h1>Página "about"</h1>
</div>
`;

exports[`Tests de vista 'Contact' Test1 - Snapshot vista 'Contact' 1`] = `
<div class="contact">
  <h1>Página "contacto"</h1>
</div>
`;
